crates/chat-cli/src/cli/chat/cli/todos.rs

```
@@ -0,0 +1,254 @@
+                Ok(entries) => {
+                    if entries.is_empty() {
+                        execute!(session.stderr, style::Print("No to-do lists to delete!\n"))?;
+                    } else if let Some(index) = fuzzy_select_todos(&entries, "Select a to-do list to delete:") {
+                        if index < entries.len() {
+                            match os.database.delete_todo(&entries[index].id) {
+                                Ok(_) => {},
+                                Err(_) => {
+                                    return Err(ChatError::Custom("Could not delete the selected to-do list".into()));
+                                },
+                            };
+                            execute!(
+                                session.stderr,
+                                style::Print("✔ Deleted to-do list: ".green()),
+                                style::Print(format!("{}\n", entries[index].description.clone().dark_grey()))
+                            )?;
+                        }
+                    }
+                },
+                Err(_) => return Err(ChatError::Custom("Could not show to-do lists".into())),
+            },
+            Self::Show => {
+                if let Some(id) = TodoState::get_current_todo_id(os).unwrap_or(None) {
+                    let state = match TodoState::load(os, &id) {
+                        Ok(s) => s,
+                        Err(_) => {
+                            return Err(ChatError::Custom("Could not load current to-do list".into()));
+                        },
+                    };
+                    match state.display_list(&mut session.stderr) {
+                        Ok(_) => execute!(session.stderr, style::Print("\n"))?,
+                        Err(_) => {
+                            return Err(ChatError::Custom("Could not display current to-do list".into()));
+                        },
+                    };
```

Do we want to propagate the error here as well. 

Also a less verbose way of writing this would be: 
```rust 
let state = TodoSate::load(os, &id).map_err(|_| ChatError::Custom("Could not load current to-do list".into())?;
```